wgpu-core/src/instance.rs

```diff
@@ -579,9 +583,18 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
         &self,
         adapter_id: AdapterId,
         desc: &DeviceDescriptor,
+        unsafe_extensions: UnsafeExtensions,
```

Comment:
Actually, let's lift this up to `request_adapter`.
First, because it already has one native-specific parameter `backends`, so adding another one doesn't make it less consistent.
Secondly, because the device has to be created with extensions exposed by an adapter anyway. So, if we lift this to adapter request, and the user doesn't give the permission, then the adapter simply doesn't expose any unsafe extensions.

Reply:
I think that's reasonable, though we would also need to have it as a parameter on the various list/choose adapter functions would likely need to be stored as part of the adapter struct itself.

This also makes me realize we never actually validated that the extensions that are requested are a subset of what the adapter exposed